core/java/iesi-core/src/main/java/io/metadew/iesi/connection/operation/ConnectionOperation.java

```diff
@@ -70,15 +73,15 @@ public HostConnection getHostConnection(Connection connection) {
             for (Map.Entry<String, ConnectionTypeParameter> connectionTypeParameter : connectionType.getParameters().entrySet()) {
                 if (connectionTypeParameter.getValue().isEncrypted()) {
                     if (connectionTypeParameter.getKey().equalsIgnoreCase("host")) {
-                        hostName = FrameworkCrypto.getInstance().decrypt(hostName);
+                        hostName = FrameworkCrypto.getInstance().decryptIfNeeded(hostName);
                     } else if (connectionTypeParameter.getKey().equalsIgnoreCase("temppath")) {
-                        tempPath = FrameworkCrypto.getInstance().decrypt(tempPath);
+                        tempPath = FrameworkCrypto.getInstance().decryptIfNeeded(tempPath);
                     }
                 }
             }
 
             hostConnection = new WindowsHostConnection(hostName, tempPath);
-        } else if (connection.getType().equalsIgnoreCase("host.linux")) {
+        } else if (connection.getType().equalsIgnoreCase("host.linux") || connection.getType().equalsIgnoreCase("host.unix")) {
```

Comment:
Not better to do connection.getType().matches("host.linux|host.unix") for more readability ?

Reply:
not in my opinion, also we will have to include the case insensitivity in the regex pattern. A regex is more appropriate when the pattern is more complex or generic. Like yours for the 2XX check the OpenAPI PR

Reply:
Indeed, my bad